pkg/capabilities/oci/manifest/manifest_test.go

```diff
@@ -0,0 +1,154 @@
+		MediaType: mediaType,
+		Manifests: []specs.Descriptor{{
+			MediaType:   specs.MediaTypeDescriptor,
+			Digest:      digest.FromString("mydummydigest"),
+			Size:        1024,
+			URLs:        []string{"ghcr.io/kubewarden/policy-server:latest"},
+			Annotations: map[string]string{"annotation": "value"},
+			Platform: &specs.Platform{
+				Architecture: "amd64",
+				OS:           "linux",
+			},
+		},
+		},
+		// Annotations contains arbitrary metadata for the image index.
+		Annotations: map[string]string{"annonation": "annotationValue"},
+	}
+}
+
+func TestV1OciManifest(t *testing.T) {
+
+	tests := []struct {
+		manifest                     interface{}
+		failsBecauseUnknownMediaType bool
+	}{
+		{
+			manifest:                     buildManifest(specs.MediaTypeImageManifest),
+			failsBecauseUnknownMediaType: false,
+		},
+		{
+			manifest:                     buildManifest(constants.ImageManifestMediaType),
+			failsBecauseUnknownMediaType: false,
+		},
+		{
+			manifest:                     buildIndexManifest(specs.MediaTypeImageIndex),
+			failsBecauseUnknownMediaType: false,
+		},
+		{
+			manifest:                     buildIndexManifest(constants.ImageManifestListMediaType),
+			failsBecauseUnknownMediaType: false,
+		},
+		{
+			manifest:                     buildIndexManifest(specs.MediaTypeDescriptor),
+			failsBecauseUnknownMediaType: true,
+		},
+		{
+			manifest:                     buildIndexManifest(specs.MediaTypeImageConfig),
+			failsBecauseUnknownMediaType: true,
+		},
+		{
+			manifest:                     buildIndexManifest(specs.MediaTypeImageLayerGzip),
+			failsBecauseUnknownMediaType: true,
+		},
+		{
+			manifest:                     buildManifest(specs.MediaTypeDescriptor),
+			failsBecauseUnknownMediaType: true,
+		},
+		{
+			manifest:                     buildManifest(specs.MediaTypeImageConfig),
+			failsBecauseUnknownMediaType: true,
+		},
+		{
+			manifest:                     buildManifest(specs.MediaTypeImageLayerGzip),
+			failsBecauseUnknownMediaType: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			manifestPayload, err := json.Marshal(test.manifest)
+			if err != nil {
+				t.Fatalf("cannot serialize response object: %v", err)
+			}
+			expectedPayload := `"myimage:latest"`
+			host := buildHostMock(t, []byte(expectedPayload), manifestPayload)
+
+			res, err := GetOCIManifest(host, "myimage:latest")
+
+			if test.failsBecauseUnknownMediaType {
+				if err == nil || !strings.Contains(err.Error(), "not a valid media type") {
+					t.Fatal("expected an invalid media type error")
+				}
+				return
+			}
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			switch test.manifest.(type) {
+			case specs.Manifest:
+				if res.IndexManifest() != nil {
+					t.Fatal("response should not be an index manifest")
+				}
+				if diff := cmp.Diff(*res.ImageManifest(), test.manifest); diff != "" {
+					t.Fatalf("unexpected manifest:\n%s", diff)
+				}
```

Comment:
the `cmp.Diff` takes 2 interfaces as variables. You can move this test outside of the switch statement and write it once:

```golang
var manifest {}interface
switch test.manifest.(type) {
case specs.Manifest:
   manifest = res.IndexManifest()
case specs.Index:
   manifest = res.ImageManifest()
default:
   // boom
}

if diff := cmp.Diff(manifest, test.manifest); diff != "" {
  t.Fatalf("unexpected manifest:\n%s", diff)
}
```